### PR TITLE
VS-1335 bgzip on extract ah var store edition

### PR DESCRIPTION
**Pull request by gbggrant**
Optionally extract VCFs in bgzipped compression format.

Integration test - [Here](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Integration/job_history/60357098-e6f2-4264-a322-c3088ce36152)

[Here](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Quickstart%20v3%20ggrant/job_history/2c4dea90-367f-4875-939a-ce0b9ae296e7) is an example extract using bgzip
[Here](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Quickstart%20v3%20ggrant/job_history/7a8413da-499b-4f93-ba85-554937249bd4) is an example extract not using bgzip (so, just '.gz')

### CODE REVIEW

File: scripts/variantstore/wdl/GvsExtractCallset.wdl [hunk 1]
```diff
@@ -20,6 +20,7 @@ workflow GvsExtractCallset {
     Int? scatter_count
     Int? extract_memory_override_gib
     Int? disk_override
+    Boolean bgzip_output_vcfs = true
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsExtractCallset.wdl:23]: The ticket says this should be off by default, although if we're running this WDL singly we're *probably* doing an AoU "small callset" where we actually do want this on...

So I'm not sure if we should leave this as you have it, or default to `false` and update the (currently nonexistent) AoU small callset docs to specify that this should be set to `true`?

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl [hunk 2]
```diff
@@ -31,6 +31,7 @@ workflow GvsExtractCohortFromSampleNames {
     String? output_gcs_dir
     # set to "NONE" if all the reference data was loaded into GVS in GvsImportGenomes
     String drop_state = "NONE"
+    Boolean bgzip_output_vcfs = true
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:34]: same concerns here as per my earlier comment

File: scripts/variantstore/wdl/GvsExtractCallset.wdl (hunk 1 above)
Comment (gbggrant) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:23]: Thanks. That's what I was thinking. That is, have it off by default, but enable it here for the likely AoU small call sets case. I agree that it should be off by default.

File: scripts/variantstore/wdl/GvsQuickstartIntegration.wdl [hunk 3]
```diff
@@ -75,6 +75,7 @@ workflow GvsQuickstartIntegration {
     # necessarily the same as the branch name selected in Terra for the integration `GvsQuickstartIntegration` workflow,
     # though in practice likely they are the same.
     if (run_hail_integration) {
+        # This workflow will be the *closest* to AoU test.
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsQuickstartIntegration.wdl:78]: nit: pretty sure I know what you mean but I am struggling to parse this sentence

**Review by mcovarr — APPROVED** [review]

File: scripts/variantstore/wdl/GvsQuickstartIntegration.wdl [hunk 4]
```diff
@@ -5,6 +5,8 @@ import "GvsQuickstartHailIntegration.wdl" as QuickstartHailIntegration
 import "GvsJointVariantCalling.wdl" as JointVariantCalling
 import "GvsUtils.wdl" as Utils
 
+# This is a comment!
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsQuickstartIntegration.wdl:8]: While factually correct, we might want to get rid of this.

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 5]
```diff
@@ -92,6 +92,7 @@
         - This workflow does not use the Terra Data Entity Model to run, so be sure to select the `Run workflow with inputs defined by file paths` workflow submission option.
     - Specify the same `call_set_identifier`, `dataset_name`, `project_id`, `extract_table_prefix`, and `interval_list` that were used in the `GvsPrepareRangesCallset` run documented above.
     - Specify the `interval_weights_bed` appropriate for the PGEN / VCF extraction run you are performing. `gs://gvs_quickstart_storage/weights/gvs_full_vet_weights_1kb_padded_orig.bed` is the interval weights BED used for Quickstart.
+    - For `GvsExtractCallset` only, set the `bgzip_output_vcfs` input to `true` to generate VCFs that are compressed using bgzip.
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:95]: I'm not sure what this means.  Are there other single-step workflows that have this input?  I think "If you want to output VCFs that are compressed using bgzip, set the `GvsExtractCallset` `bgzip_output_vcfs` input to `true`."

**Review by rsasch — CHANGES_REQUESTED** [review]
One small change/clarification, plus I agree with Miguel's comments about comments.

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md (hunk 5 above)
Comment (mcovarr) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:95]: This is  is under the section that covers both `GvsExtractCallset` and `GvsExtractCallsetPgenMerged` so I *think* this is saying that these remarks apply only to the former.

**Review by mcovarr — APPROVED** [review]

**Review by rsasch — APPROVED** [review]
If Miguel doesn't care that his comments about comments weren't addressed, I guess I don't either.

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md (hunk 5 above)
Comment (gbggrant) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:95]: Yes, that's what I meant. I will update the documentation to make it clearer.

**Pull request merged**